modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java

```diff
@@ -325,9 +325,11 @@ public String toDefaultValue(Schema p) {
         } else if (ModelUtils.isDateTimeSchema(p)) {
             return UNDEFINED_VALUE;
         } else if (ModelUtils.isNumberSchema(p)) {
-            NumberSchema dp = (NumberSchema) p;
-            if (dp.getDefault() != null) {
-                return dp.getDefault().toString();
+            if  (p instanceof  NumberSchema ){
+                NumberSchema dp = (NumberSchema) p;
+                if (dp.getDefault() != null) {
+                    return dp.getDefault().toString();
+                }
```

Comment:
Shall we instead follow the case for Integer below without any doing any casting?
```
             if (p.getDefault() != null) {
                 return p.getDefault().toString();
             }
```
Line 336 to 338 after the change

Reply:
@wing328 yes indeed 
if it fix 
error and it is less code 
It is better ;-)

Reply:
👌 I'll merge this PR after you update the fix.